File: java/org.financer.client/src/main/java/org/financer/client/connection/error/ServerNotAvailableException.java
package org.financer.client.connection.error;

import org.financer.client.connection.RequestConfig;
import org.financer.shared.exceptions.FinancerException;

public class ServerNotAvailableException extends FinancerException {

    private static final String MESSAGE = "Request [%s: %s] failed. Server is not probably available.";
    private static final String MESSAGE_KEY = "exception.serverNotAvailable";

    public ServerNotAvailableException(RequestConfig requestConfig) {
        super(MESSAGE, MESSAGE_KEY, requestConfig.getHttpMethod().toString(), requestConfig.getPath());
    }
}

File: java/org.financer.client/src/main/kotlin/org/financer/client/connection/error/ServerNotAvailableException.kt
package org.financer.client.connection.error

import org.financer.client.connection.RequestConfig
import org.financer.shared.exceptions.FinancerException

class ServerNotAvailableException(requestConfig: RequestConfig) : FinancerException(MESSAGE, MESSAGE_KEY, requestConfig.httpMethod.toString(), requestConfig.path) {
    companion object {
        private const val MESSAGE = "Request [%s: %s] failed. Server is not probably available."
        private const val MESSAGE_KEY = "exception.serverNotAvailable"
    }
}